### PR TITLE
Deep collection structures

### PR DESCRIPTION
**Pull request by andykog**
This PR allows creating mutable collection of collections of collections of collections..., adds basic methods to work with deep values by using index path (simple Array<Int>, instead of NSIndexPath, as the last one is not so safe and convenient). All the previous functionality is kept, but the API is slightly changed. (`CollectionChange` became `FlatMutableCollectionChange`, `changesSignal` became `flatChangesSignal`...)

### CODE REVIEW

File: MutableCollectionProperty/Array+diff.swift [hunk 1]
```diff
@@ -0,0 +1,55 @@
+//
+//  Array+diff.swift
+//  MutableCollectionProperty
```
Comment (pepibumur) [inline MutableCollectionProperty/Array+diff.swift]: Let's remove the headers from Swif tfiles

File: MutableCollectionProperty/MutableCollectionSection.swift [hunk 2]
```diff
@@ -0,0 +1,106 @@
+//
+//  MutableCollectionSection.swift
+//  MutableCollectionProperty
```
Comment (pepibumur) [inline MutableCollectionProperty/MutableCollectionSection.swift]: The same here

File: MutableCollectionPropertyTests/MutableCollectionPropertyTests.swift [hunk 3]
```diff
@@ -269,192 +196,224 @@ class MutableCollectionPropertyTests: QuickSpec {
...
+                            }
+                        }
+                        property.insert("test0", atIndexPath: [0, 0])
+                    })
+                }
+                
+            })
+            
+            context("replacing elements", {
+                // TODOD
```
Comment (pepibumur) [inline MutableCollectionPropertyTests/MutableCollectionPropertyTests.swift]: Missing tests

**Comment by pepibumur** [comment]
Great job @andykog :clap: . Could you add an small note in the README with how to use this support to deep collections?

File: MutableCollectionPropertyTests/MutableCollectionPropertyTests.swift (hunk 3 above)
Comment (andykog) [inline MutableCollectionPropertyTests/MutableCollectionPropertyTests.swift]: Those features are not done yet (and I'm not shure they are usefull in context of deep structure)

File: MutableCollectionProperty/Array+diff.swift (hunk 1 above)
Comment (andykog) [inline MutableCollectionProperty/Array+diff.swift]: Done

File: MutableCollectionProperty/MutableCollectionSection.swift (hunk 2 above)
Comment (andykog) [inline MutableCollectionProperty/MutableCollectionSection.swift]: Done

**Comment by andykog** [comment]
> Could you add an small note in the README
Working on it

**Comment by andykog** [comment]
I was to hastily making PR, need more time for testing and improvements

**Comment by pepibumur** [comment]
Sure, take your time @andykog

**Comment by andykog** [comment]
It seems to be ok now.

**Comment by pepibumur** [comment]
What's the status of this @andykog ?

**Comment by andykog** [comment]
You will laugh, but I'm still working :-)

**Comment by andykog** [comment]
Now it's done

**Comment by andykog** [comment]
Updated for swift 2.3 compatibility + added minor improvement

**Comment by pepibumur** [comment]
@andykog I didn't have much time these days to review your PR. I granted you write permissions on this repository so your'e now contributor of it. Feel free to merge the PRs and version the library. Just remember:
- Having CI passing.
- Once the branch is merged, create a new release in GitHub attaching the Carthage binary and update the podspec version as well. If you need permissions to push the podspec to CocoaPods repository let me know your email and I can include you as a pod owner.

You did a great job here. I'm very impressed about what you managed to do. Thanks for this contribution.

**Comment by andykog** [comment]
Thanks @pepibumur, my email is mail@andykog.com

**Comment by pepibumur** [comment]
Can't add you as owner, do you have any other pod contribution with that email @andykog ?

**Comment by andykog** [comment]
Oh, you're right, I have just now registered in pod trunk

**Comment by pepibumur** [comment]
Done @andykog ! try with this version and versioning it :stuck_out_tongue:

**Comment by andykog** [comment]
Ok, thanks :-)

**Pull request merged**